src/model/state.mc

```diff
@@ -0,0 +1,28 @@
+-- Represents a state. Meant for use with DFA
+
+-- It is formed of the state type which is user-defined and two functions: one for printing the state and one for equality checks.
+
+
+type STATE = {
+     state2tring: fun_2string,
+     eqs: fun_eq,
```

Comment:
Typo here: `state2tring`
I'm guessing the type signatures should be `a -> String` and `a -> a -> Bool`